### ISSUE TITLE
undefined method `install_gem' for #<Gem::Commands::InstallCommand:0x007ff878992ed8> (NoMethodError)

### ISSUE THREAD

**Issue by kballard** [fastlane/fastlane#11413]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
`fastlane` throws an error on macOS 10.12 when it tries to install the `xcode-install` gem.

##### Complete output when running fastlane, including the stack trace and command used
```
fastlane ios xarth_fabric_external_qa
[⠼] 🚀 /Library/Ruby/Gems/2.0.0/gems/fastimage-2.1.1/lib/fastimage.rb:68: warning: already initialized constant URI::DEFAULT_PARSER
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/uri/common.rb:545: warning: previous definition of DEFAULT_PARSER was here
[✔] 🚀 
[14:02:13]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[14:02:13]: ---------------------------
[14:02:13]: --- Step: opt_out_usage ---
[14:02:13]: ---------------------------
[14:02:13]: Disabled upload of used actions
[14:02:13]: -------------------------------------------------
[14:02:13]: --- Step: Verifying required fastlane version ---
[14:02:13]: -------------------------------------------------
[14:02:13]: Your fastlane version 2.72.0 matches the minimum requirement of 2.39.2  ✅
[14:02:13]: ------------------------------
[14:02:13]: --- Step: default_platform ---
[14:02:13]: ------------------------------
[14:02:13]: Driving the lane 'ios xarth_fabric_external_qa' 🚀
[14:02:13]: ----------------------------------
[14:02:13]: --- Step: ensure_xcode_version ---
[14:02:13]: ----------------------------------
[14:02:14]: Installing Ruby gem 'xcode-install'...
[14:03:14]: Sending crash report...
[14:03:14]: The stack trace is sanitized so no personal information is sent.
[14:03:14]: To see what we are sending, look here: /Users/eridius/.fastlane/latest_crash.json
[14:03:14]: Learn more at https://github.com/fastlane/fastlane#crash-reporting
[14:03:14]: You can disable crash reporting by adding `opt_out_crash_reporting` at the top of your Fastfile
+------------------+------------------------------+
|                  Lane Context                   |
+------------------+------------------------------+
| DEFAULT_PLATFORM | ios                          |
| PLATFORM_NAME    | ios                          |
| LANE_NAME        | ios xarth_fabric_external_qa |
+------------------+------------------------------+
[14:03:14]: undefined method `install_gem' for #<Gem::Commands::InstallCommand:0x007ff878992ed8>

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | opt_out_usage                       | 0           |
| 2    | Verifying required fastlane version | 0           |
| 3    | default_platform                    | 0           |
| 💥   | ensure_xcode_version                | 59          |
+------+-------------------------------------+-------------+

[14:03:14]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  fastlane beta- undefined method `install_gem' for #<Gem::Commands::InstallCommand:0x007fcf2b4d7900>
    https://github.com/fastlane/fastlane/issues/8290 [closed] 7 💬
    07 Aug 2017

➡️  undefined method `install_gem' for #<Gem::Commands::InstallCommand
    https://github.com/fastlane/fastlane/issues/8489 [closed] 2 💬
    09 Jun 2017

➡️  undefined method `install_gem' for #<Gem::Commands::InstallCommand:0x007ffb208656a0>
    https://github.com/fastlane/fastlane/issues/7901 [closed] 6 💬
    19 May 2017

and 8 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60install_gem%27%20for%20Gem::Commands::InstallCommand&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/fastlane_require.rb:36:in `install_gem_if_needed': [!] undefined method `install_gem' for #<Gem::Commands::InstallCommand:0x007ff878992ed8> (NoMethodError)
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/helper/gem_helper.rb:7:in `verify_gem!'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/actions/ensure_xcode_version.rb:9:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:231:in `block in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
	from Fastfile:36:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:287:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:287:in `execute_flow_block'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:46:in `block in execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/command_line_handler.rb:34:in `handle'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:66:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/commands_generator.rb:343:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/fastlane/lib/fastlane/cli_tools_distributor.rb:91:in `take_off'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-2.72.0/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.12.6                                                        |
| Ruby                        | 2.0.0                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.14.3 (Apple Git-98)                              |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.12.6 (16G1114)                                     |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib |
| OpenSSL Version             | OpenSSL 0.9.8zc 19 Mar 2015                                    |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 9.2                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

(file trimmed to relevant bit)

```ruby
fastlane_version "2.39.2"

default_platform :ios

platform :ios do
  before_all do
    ensure_xcode_version(version: "9.2")
    # rest of file trimmed
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.72.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.7.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.60.0       |
| faraday                   | 0.13.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.3        |
| json                      | 1.7.7        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.2       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| libxml-ruby               | 2.6.0        |
</details>

*generated on:* **2018-01-03**
</details>

**Comment by KrauseFx** [comment]
Hey @kballard, thanks for the report. That's interesting, could you try updating your RubyGems system environment?

First please run
```
gem -v
```
and share the output with us, so that we can fix it 👍 

then run
```
gem install rubygems-update
update_rubygems
gem update --system
```
^ The above commands might require `sudo`, depending on your setup 👍

**Comment by kballard** [comment]
`gem -v` prints `2.0.14.1`. I should be using the stock RubyGems that ships with macOS 10.12; I don't recall ever updating rubygems on this machine. I'll try updating it in a minute.

**Comment by kballard** [comment]
Looks like `update_rubygems` is not possible to run on the stock rubygems that ships with the system. Even with `sudo` I get `Errno::EPERM`, so I imagine that `/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/gem` is protected by SIP.

**Comment by kballard** [comment]
I also just got a similar error at the very end of a fastlane build (I installed `xcode-install` manually to work around the initial error) when trying to install another gem:

> uninitialized constant Gem::Resolver

**Comment by kballard** [comment]
Although I did try running the rubygems update stuff while the fastlane build was going on, which may not have been a good idea. In any case, I ran `gem update --system` and `gem -v` now prints `2.7.4`. I don't know what `update_rubygems` was supposed to do, since I couldn't run it.

**Comment by kballard** [comment]
Uninstalling `xcode-install` and re-running `fastlane` no longer throws the error (now that I have gem 2.7.4). Of course, it throws a different error, which is that installing gems requires `sudo` >_< And of course I don't want to run all of `fastlane` with sudo. But that's a different issue.

**Comment by kballard** [comment]
Oh dear. `sudo gem install xcode-install` now tries to install the `xcversion` binary in `/usr/bin` (which is blocked by SIP) instead of `/usr/local/bin`. Apparently updating rubygems actually just broke my rubygems installation such that I can't install binaries anymore >_<.

Any idea how to fix this?
